vowpalwabbit/core/src/reductions/plt.cc

```diff
@@ -293,48 +314,38 @@ void print_update_plt(VW::workspace& all, VW::shared_data&, const plt&, const VW
 
 void finish(plt& p)
 {
-  // print results in test mode
+  // print results in the test mode
   if (!p.all->training && p.ec_count > 0)
   {
     // top-k predictions
     if (p.top_k > 0)
     {
-      double correct = 0;
       for (size_t i = 0; i < p.top_k; ++i)
       {
-        correct += p.tp_at[i];
         // TODO: is this the correct logger?
-        *(p.all->trace_message) << "p@" << i + 1 << " = " << correct / (p.ec_count * (i + 1)) << std::endl;
-        *(p.all->trace_message) << "r@" << i + 1 << " = " << correct / p.true_count << std::endl;
+        *(p.all->trace_message) << "p@" << i + 1 << " = " << p.p_at[i] / p.ec_count << std::endl;
+        *(p.all->trace_message) << "r@" << i + 1 << " = " << p.r_at[i] / p.ec_count << std::endl;
       }
     }
-
     else if (p.threshold > 0)
     {
       // TODO: is this the correct logger?
       *(p.all->trace_message) << "hamming loss = " << static_cast<double>(p.fp + p.fn) / p.ec_count << std::endl;
-      *(p.all->trace_message) << "precision = " << static_cast<double>(p.tp) / (p.tp + p.fp) << std::endl;
-      *(p.all->trace_message) << "recall = " << static_cast<double>(p.tp) / (p.tp + p.fn) << std::endl;
+      *(p.all->trace_message) << "micro-precision = " << static_cast<double>(p.tp) / (p.tp + p.fp) << std::endl;
+      *(p.all->trace_message) << "micro-recall = " << static_cast<double>(p.tp) / (p.tp + p.fn) << std::endl;
     }
   }
 }
 
 void save_load_tree(plt& p, io_buf& model_file, bool read, bool text)
 {
+  std::stringstream msg;
   if (model_file.num_files() > 0)
   {
-    bool resume = p.all->save_resume;
-    std::stringstream msg;
-    msg << ":" << resume << "\n";
-    bin_text_read_write_fixed(model_file, reinterpret_cast<char*>(&resume), sizeof(resume), read, msg, text);
-
-    if (resume && !p.all->weights.adaptive)
+    for (size_t i = 0; i < p.t; ++i)
     {
-      for (size_t i = 0; i < p.t; ++i)
-      {
-        bin_text_read_write_fixed(
-            model_file, reinterpret_cast<char*>(&p.nodes_time[i]), sizeof(p.nodes_time[0]), read, msg, text);
-      }
+      bin_text_read_write_fixed(
+          model_file, reinterpret_cast<char*>(&p.nodes_time[i]), sizeof(p.nodes_time[0]), read, msg, text);
```

Comment:
It looks like this was changed to not use the `resume` value any more. To avoid changing the binary format of the model we can read/write this value but just ignore it.

Reply:
@jackgerrits the small modification of this function is needed, cause the previous one has a bug that makes models load incorrectly (when resume is true). I've modified it to read/write resume variable as you suggested. I've tested it and it works correctly now with previous versions.